### ISSUE TITLE
Expected a script name instead found

### ISSUE THREAD

**Issue by julemand101** [bsutton/dshell#40]
This is most likely a usability problem but I think it is important to do something about it.

The first thing I wanted to do after installing dshell was to create a new problem which I did:
```
[julemand101@beta ~]$ mkdir dshell
[julemand101@beta ~]$ cd dshell/
[julemand101@beta dshell]$ dshell create hello
Creating project.
Created Virtual Project at /home/julemand101/.dshell/cache/home/julemand101/dshell/hello.project
Running pub get...
Resolving dependencies...
+ args 1.5.2
+ charcode 1.1.2
+ collection 1.14.12
+ dshell 1.0.38
+ equatable 1.0.2
+ file 5.1.0
+ file_utils 0.1.4
+ globbing 0.3.0
+ intl 0.16.1
+ logger 0.8.2
+ matcher 0.12.6
+ meta 1.1.8
+ money2 1.3.0
+ path 1.6.4
+ pub_semver 1.4.2
+ pubspec 0.1.3
+ quiver 2.1.2+1
+ recase 3.0.0
+ source_span 1.5.5
+ stack_trace 1.9.3
+ string_scanner 1.0.5
+ term_glyph 1.1.0
+ uri 0.11.3+1
+ utf 0.9.0+5
+ yaml 2.2.0
Changed 25 dependencies!
Precompiling executables...
Precompiled dshell:dshell_complete.
Precompiled dshell:dshell.
Making script executable
Project creation complete.
To run your script:
   ./hello
```

But when I try using the suggested way to run the script I ends up with:
```
[julemand101@beta dshell]$ ./hello
dshell: Expected a script name instead found: ./hello

For help with dshell options:
  dshell help | help <command>
    Displays the usage message | Display the command's usage message.
```

After checking the source code it makes totally sense this error happens because of the following check for the script are ending with ".dart": 
https://github.com/bsutton/dshell/blob/03d3d7275a2f4e6b3283efce401d8b7c99e7fa6f/lib/src/script/script.dart#L97-L100

But I think that if this requirement is important, then we "create" command should either stop me from creating a non-working script or automatically suffix ".dart" to the name of the script. 

Personally, I think it should be consider if the requirement of ".dart" makes sense since you often (on Linux) don't have a file extension for your scripts since the interpreter is determined by the shebang tag.

**Comment by bsutton** [comment]
I came across the create bug just yesterday and it's on my to-do list. But
thanks for raising the issue.

So you are arguing that we should relax the requirement for all scripts to
end in .dart?

Have you looked at they compile command?

It currently assume that the script ends in .dart but the executable has no
extension.

Do you have thoughts on how the complie should work of we remove the
extension requirement?

I've also considered changing the extension to .ds.

It's shorter, and differentiates dshell scripts.

The down side to this is that ides wouldn't know how to do syntax
highlighting which would be quite detrimental.

This might actually be the strongest argument for enforcing the .dart.

**Comment by bsutton** [comment]
I've just published 1.0.39 which now enforces the .dart extension when creating scripts.
I think the IDE issue is significant enough that we should stick to enforcing the extensions.

I will close this issue but if you disagree feel free to re-open the issue and make your arguments :)
